### PR TITLE
move metrics from settings to client classes

### PR DESCRIPTION
**Pull request by fidelcoria-aa**
**Affects:** all versions.

---

When configuring the sonar instances from admin page in UI the entries change order. So the metrics in properties file do not match the order of the UI admin page. This PR moves the metrics from properties file to client classes.

### CODE REVIEW

**Comment by Courtneyp123** [comment]
Hi @fidelcoria-aa could you explain why you are moving metrics from properties file to the class?

**Comment by fidelcoria-aa** [comment]
Sure
**Problematic situation:**
Suppose you have 2 sonar instances SonarA and SonarB. You go to Admin UI Page to enter Sonar details (url, username, password). While other information is added to properties file (version, metrics, etc.) SonarA has index 0 and SonarB has index 1.
*Now* notice the order of values in properties file is fixed, when you update SonarA password and click save... the order in the database changes. So SonarA has index 1 now and starts using the wrong properties file vales. (This seems to be a bug with Mongo or Mongo Driver, I was not able to pin point problem)

**Fix:**
Fetch version every time and Do not store version (because the order changes as discussed above). And same applied for the Metrics fields

**Comment by Courtneyp123** [comment]
Great, thanks for the clarification.  I noticed that this PR doesn't make use of PR #5 , specifically getSonarVersion(instanceUrl) to collect versions from the url.  Were you planning on implementing that back in so that the version still fetches from server?

**Comment by fidelcoria-aa** [comment]
Well, once #5 is merged to master. Then #6 can be merged into master if no conflicts. If there are conflicts then I can resolve conflicts

File: src/main/java/com/capitalone/dashboard/collector/DefaultSonar6Client.java [hunk 1]
```diff
@@ -154,9 +152,9 @@ private void getProjects(String url, String key, JSONArray jsonArray, int pageNu
     }
 
     @Override
-    public CodeQuality currentCodeQuality(SonarProject project, String metrics) {
+    public CodeQuality currentCodeQuality(SonarProject project) {
         String url = String.format(
-                project.getInstanceUrl() + URL_RESOURCE_DETAILS, project.getProjectId(), metrics);
+                project.getInstanceUrl() + URL_RESOURCE_DETAILS, project.getProjectId(), METRICS);
```
Comment (rvema) [inline src/main/java/com/capitalone/dashboard/collector/DefaultSonar6Client.java]: can we add a check to see if metrics passed is null or empty and then use the default metrics?

Comment (rvema) [reply src/main/java/com/capitalone/dashboard/collector/DefaultSonar6Client.java]: we want the flexibility of loading the metrics from the properties as well.

Comment (fidelcoria-aa) [reply src/main/java/com/capitalone/dashboard/collector/DefaultSonar6Client.java]: Hmm, ok. I think the metrics property would be best as a "global" sonar property (sonar.metrics6=blah,blah) but not instance specific (sonar.metrics[2]=quux,ducks). is that fair? and you're saying if metrics are in properties they override the client METRICS right?

Comment (fidelcoria-aa) [reply src/main/java/com/capitalone/dashboard/collector/DefaultSonar6Client.java]: ok, I'll have something worked out by tomorrow

**Comment by Courtneyp123** [comment]
@fidelcoria-aa there seems to be an issue with the new changes collecting metrics63andAbove and metricsBefore63.  They are null when the properties files have metrics.

**Comment by fidelcoria-aa** [comment]
Hmm. I wonder if it has to do with changing the metrics type from List<String> to String. I'll take a look

**Comment by fidelcoria-aa** [comment]
I tested with sonar instance version 6.0 and version 6.7.2 and it worked as expected both times. 
I also did a full integration test with 6.7.2: I enabled a sonar project on a dashboard and ran the collector and it used the metrics string from properties file.

**Comment by rvema** [comment]
Thanks Fidel, Courtney will test it with 5.6 version and get back

**Pull request merged**